roles/robot-pkgs/vars/main.yml

```diff
@@ -3,12 +3,14 @@
 
 # Keep UMD first to prefer downloads from there
 ros_repos:
-  - deb http://mirror.umd.edu/packages.ros.org/ros/ubuntu xenial main
-  - deb http://packages.ros.org/ros/ubuntu xenial main
+  - deb http://mirror.umd.edu/packages.ros.org/ros/ubuntu bionic main
+  - deb http://packages.ros.org/ros/ubuntu bionic main
```

Comment:
There should be an `ubuntu_release` variable that gets registered in `common`. Can we change this to use a Jinja2 template and then put this into `/etc/apt/sources.list.d/ros-packages.list`? The template would go in `roles/robot-pkgs/templates/ros-packages.list.j2` and the files in `roles/common/templates/*.list` would be a good template (but clearly only need these two repos, not the full suite of Ubuntu/Mint repos).

I'd also like to see the mirrors enumerated in the `vars/main.yml` file. So ideally, something like:

```
{%for mirror in ros_mirrors %}
deb {{ mirror_url }} {{ ubuntu_release }} main
{% endfor %}
```

but the syntax, etc will need to be verified/tested.

This would also require changes to `tasks/main.yml` to no longer use the `apt_repository` module and instead use the `template` module.

Reply:
This will definitely eliminate having to change the release name every time. See  commit 91b364e for the Jinja2 template